go.mod

```diff
@@ -4,49 +4,39 @@ go 1.16
 
 require (
 	cloud.google.com/go v0.75.0
-	github.com/blang/semver v3.5.1+incompatible // indirect
 	github.com/go-openapi/spec v0.19.6 // indirect
 	github.com/go-openapi/swag v0.19.7 // indirect
 	github.com/gofrs/flock v0.7.1
-	github.com/gogo/protobuf v1.3.2 // indirect
-	github.com/google/go-cmp v0.5.4
+	github.com/google/go-cmp v0.5.5
 	github.com/google/go-tpm v0.2.0
 	github.com/google/uuid v1.1.4 // indirect
 	github.com/hashicorp/golang-lru v0.5.4 // indirect
 	github.com/imdario/mergo v0.3.11 // indirect
-	github.com/mrunalp/fileutils v0.5.0 // indirect
 	github.com/onsi/ginkgo v1.14.1 // indirect
 	github.com/onsi/gomega v1.10.3 // indirect
-	github.com/pkg/errors v0.9.1 // indirect
 	github.com/prometheus/client_golang v1.7.1
-	github.com/prometheus/client_model v0.2.0 // indirect
-	github.com/spf13/cobra v1.1.1
+	github.com/spf13/cobra v1.1.3
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
-	go.etcd.io/etcd v0.5.0-alpha.5.0.20200910180754-dd1b699fc489 // indirect
 	go.uber.org/zap v1.14.1 // indirect
-	golang.org/x/lint v0.0.0-20201208152925-83fdc39ff7b5 // indirect
 	golang.org/x/oauth2 v0.0.0-20210112200429-01de73cf58bd
 	golang.org/x/sync v0.0.0-20201207232520-09787c993a3a // indirect
 	google.golang.org/api v0.36.0
 	google.golang.org/genproto v0.0.0-20210111234610-22ae2b108f89 // indirect
-	google.golang.org/grpc v1.34.0 // indirect
 	gopkg.in/check.v1 v1.0.0-20200902074654-038fdea0a05b // indirect
 	gopkg.in/gcfg.v1 v1.2.0
 	gopkg.in/warnings.v0 v0.1.2
-	k8s.io/api v0.21.0
-	k8s.io/apimachinery v0.21.0
-	k8s.io/apiserver v0.21.0
+	k8s.io/api v0.22.0
+	k8s.io/apimachinery v0.22.0
+	k8s.io/apiserver v0.22.0
 	k8s.io/client-go v9.0.0+incompatible
-	k8s.io/cloud-provider v0.21.0
+	k8s.io/cloud-provider v0.22.0
 	k8s.io/cloud-provider-gcp/providers v0.0.0
-	k8s.io/component-base v0.21.0
-	k8s.io/controller-manager v0.21.0
-	k8s.io/klog/v2 v2.8.0
+	k8s.io/component-base v0.22.0
+	k8s.io/controller-manager v0.22.0
+	k8s.io/klog/v2 v2.9.0
 	k8s.io/kubelet v0.21.0
 	k8s.io/kubernetes v1.21.0
```

Comment:
All the k8s `v0.21.0` modules should be upgraded to `v0.22.0` at the same time.

Reply:
When I update k8s.io/kubernetes v1.21.0 to v1.22.0, I see this:

> go: k8s.io/kubernetes@v1.22.0 requires
>	k8s.io/pod-security-admission@v0.0.0: reading k8s.io/pod-security-admission/go.mod at revision v0.0.0: unknown revision v0.0.0

How should I resolve this?

Reply:
I believe you should add `k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.22.0` in the `replace` section.

Reply:
I did this: https://github.com/openshift/cloud-provider-gcp/pull/2 It seems working.